### PR TITLE
[Admin][Product] Add validation message for the channel-specific price field

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A
|-----------------|-----
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | N/A
| License         | MIT

image

### CODE REVIEW

File: features/admin/product/managing_product_variants/generating_product_variant_validation.feature [hunk 1]
```diff
@@ -33,7 +33,7 @@ Feature: Generating product variant generation
         And I specify that the 1st variant costs "$90.00" in "United States" channel
         And I try to generate it
         Then I should be notified that code is required for the 2nd variant
-        Then I should be notified that prices in all channels must be defined for the 2nd variant
+        And I should be notified that price for the 2st variant in "United States" channel must be defined
```
Comment (TheMilek) [inline features/admin/product/managing_product_variants/generating_product_variant_validation.feature]:
```suggestion
        And I should be notified that price for the 2nd variant in "United States" channel must be defined
```

File: features/admin/product/managing_product_variants/generating_product_variant_validation.feature [hunk 2]
```diff
@@ -54,9 +54,9 @@ Feature: Generating product variant generation
         And I do not specify any information about variants
         And I try to generate it
         Then I should be notified that code is required for the 1st variant
-        And I should be notified that prices in all channels must be defined for the 1st variant
+        And I should be notified that price for the 1st variant in "United States" channel must be defined
         And I should be notified that code is required for the 2nd variant
-        And I should be notified that prices in all channels must be defined for the 2nd variant
+        And I should be notified that price for the 2st variant in "United States" channel must be defined
```
Comment (TheMilek) [inline features/admin/product/managing_product_variants/generating_product_variant_validation.feature]:
```suggestion
        And I should be notified that price for the 2nd variant in "United States" channel must be defined
```

File: features/admin/product/managing_product_variants/product_variant_validation.feature [hunk 3]
```diff
@@ -15,7 +15,7 @@ Feature: Product variant validation
         And I specify its code as "VODKA_WYBOROWA_PREMIUM"
         But I do not set its price
         And I try to add it
-        Then I should be notified that prices in all channels must be defined
+        Then I should be notified that prices in "United States" channel must be defined
```
Comment (TheMilek) [inline features/admin/product/managing_product_variants/product_variant_validation.feature:18]: I'm just wondering if we shouldn't also display the general flash message 🤔

File: src/Sylius/Behat/Page/Admin/ProductVariant/CreatePage.php [hunk 4]
```diff
@@ -107,14 +107,14 @@ protected function getDefinedElements(): array
...
             'price_calculator' => '#sylius_admin_product_variant_pricingCalculator',
             'shipping_category' => '#sylius_admin_product_variant_shippingCategory',
             'shipping_required' => '#sylius_admin_product_variant_shippingRequired',
-            'original_price' => '#sylius_admin_product_variant_channelPricings_%channelCode%_originalPrice',
-            'price' => '#sylius_admin_product_variant_channelPricings_%channelCode%_price',
+            'original_price' => '#sylius_admin_product_variant_channelPricings_%channel_code%_originalPrice',
+            'price' => '#sylius_admin_product_variant_channelPricings_%channel_code%_price',
             'prices-body' => '[data-test-product-channel-pricings-accordion-body]',
             'weight' => '#sylius_admin_product_variant_weight',
             'width' => '#sylius_admin_product_variant_width',
```
Comment (TheMilek) [inline src/Sylius/Behat/Page/Admin/ProductVariant/CreatePage.php]: it should be:
```php
'code' => '#sylius_admin_product_variant_code',
'depth' => '#sylius_admin_product_variant_depth',
'height' => '#sylius_admin_product_variant_height',
'minimum_price' => '#sylius_admin_product_variant_channelPricings_%channel_code%_minimumPrice',
'on_hand' => '#sylius_admin_product_variant_onHand',
'option_select' => '#sylius_admin_product_variant_optionValues_%option-name%',
'original_price' => '#sylius_admin_product_variant_channelPricings_%channel_code%_originalPrice',
'price' => '#sylius_admin_product_variant_channelPricings_%channel_code%_price',
'price_calculator' => '#sylius_admin_product_variant_pricingCalculator',
'prices-body' => '[data-test-product-channel-pricings-accordion-body]',
'shipping_category' => '#sylius_admin_product_variant_shippingCategory',
'shipping_required' => '#sylius_admin_product_variant_shippingRequired',
'weight' => '#sylius_admin_product_variant_weight',
'width' => '#sylius_admin_product_variant_width',
```

File: src/Sylius/Bundle/CoreBundle/Resources/translations/validators.en.yml [hunk 5]
```diff
@@ -149,6 +149,8 @@ sylius:
             min: Depth cannot be negative.
             invalid: Depth must be a number.
         channel_pricing:
+            price:
+                not_defined: 'You must define price.'
             all_defined: 'You must define price for every enabled channel.'
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Resources/translations/validators.en.yml]: Is it still in use?

File: features/admin/product/managing_product_variants/product_variant_validation.feature (hunk 3 above)
Comment (Wojdylak) [reply features/admin/product/managing_product_variants/product_variant_validation.feature:18]: I don't think it's necessary. We have decided to add a general flash message for all forms if there is an error, so I think that should be enough.

**Review by TheMilek — APPROVED** [review]

**Comment by TheMilek** [comment]
Thank you, @Wojdylak!

**Pull request merged**